src/core/Entry.cpp

```diff
@@ -274,6 +277,63 @@ const EntryAttachments* Entry::attachments() const
     return m_attachments;
 }
 
+bool Entry::hasTotp() const{
+    if (m_attributes->hasKey("otp")) {
+        // Compatibility with "KeeOtp" plugin
+        QRegExp rx("key=(.+)", Qt::CaseInsensitive, QRegExp::RegExp);
+
+        if (rx.exactMatch(m_attributes->value("otp"))) {
+            return true;
+        }
+    } else if (m_attributes->hasKey("TOTP Seed")) {
+        // Compatibility with "Tray TOTP" plugin
+        return true;
+    }
+    return false;
+}
+
+QString Entry::totp() const{
+    if (hasTotp()) {
+        QString seed = "";
+
+        if (m_attributes->hasKey("otp")) {
+            QRegExp rx("key=(.+)", Qt::CaseInsensitive, QRegExp::RegExp);
+            rx.indexIn(m_attributes->value("otp"));
+            seed = rx.cap(1);
+        } else {
+            seed = m_attributes->value("TOTP Seed");
+        }
+
+        const char *secret = qPrintable(seed.toLatin1());
+        char output[20];
+        time_t t_now = time(NULL);
+        char *result = NULL;
+        size_t result_len;
+        int rc;
+
+        rc = oath_init();
+
+        rc = oath_base32_decode(secret, strlen(secret),
+                                &result, &result_len);
+
+        if (rc == OATH_OK) {
+            rc = oath_totp_generate(result, result_len, t_now, 30, 0, 6, output);
```

Comment:
I believe this is interested in the address of the pointer container itself (hence the reference to a pointer). After this call, it will be pointing to the allocated data made in the function.

There may be a memory leak, however, because result is never deleted.